ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/preexecution/TimeBasedWriteSetSelector.scala

```diff
@@ -27,11 +27,10 @@ final class TimeBasedWriteSetSelector[ReadSet, WriteSet](now: () => Instant)
       !recordTime.isBefore(minRecordTime) &&
         !recordTime.isAfter(maxRecordTime)
     if (withinTimeBounds) {
-      val rejectionReason =
-        Rejection.RecordTimeOutOfRange(minRecordTime, maxRecordTime)
-      logger.trace(s"Transaction rejected at post-execution, ${rejectionReason.description}")
       preExecutionOutput.successWriteSet
     } else {
+      val rejectionReason = Rejection.RecordTimeOutOfRange(minRecordTime, maxRecordTime)
+      logger.trace(s"Transaction rejected at post-execution, ${rejectionReason.description}")
```

Comment:
I suggest to also add the specific reason, so that it's very clear that we are referring to this post-execution branch (b/c a success write set can also be a rejection).